lib/Dialect/TritonStructured/IR/TritonStructuredOps.cpp

```diff
@@ -137,21 +136,24 @@ GetStructuredStateOp::getOffsetAndStrideTypes(MLIRContext *context,
 }
 
 std::optional<std::pair<int32_t, int32_t>>
-GetStructuredStateOp::getOffsetAndStrideSegmentSizes(Type ptrLikeType) {
+GetStructuredStateOp::getOffsetAndStrideSegmentSizes(Type type) {
   int32_t offsetSegmentSize = 0;
   int32_t strideSegmentSize = 0;
 
-  // Unstructured pointers (tensor<!tt.ptr<type>>)
-  if (auto tensorType = llvm::dyn_cast<RankedTensorType>(ptrLikeType)) {
-    if (auto ptrType =
-            dyn_cast<triton::PointerType>(tensorType.getElementType())) {
+  if (auto tensorType = llvm::dyn_cast<RankedTensorType>(type)) {
+    if (tensorType.getElementType().isIntOrIndex()) {
```

Comment:
Ditto here, what causes us to only care about `IntOrIndex` types?

Reply:
similarly to the other comment, only tensor of int or index can potentially be used to compute pointer offsets and hence we would only need to use the `tts.get_structured_state` ops for values of such types.